src/main/java/com/immunopass/service/OrderService.java

```diff
@@ -0,0 +1,149 @@
+package com.immunopass.service;
+
+import com.immunopass.controller.OrderController;
+import com.immunopass.entity.OrderEntity;
+import com.immunopass.enums.IDType;
+import com.immunopass.enums.OrderStatus;
+import com.immunopass.enums.VoucherStatus;
+import com.immunopass.model.Order;
+import com.immunopass.model.Voucher;
+import com.immunopass.repository.OrderRepository;
+import com.immunopass.restclient.SMSService;
+import com.immunopass.utils.S3Util;
+import com.immunopass.utils.Utils;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+import java.util.UUID;
+
+import static com.immunopass.utils.ParameterCheckUtils.*;
+
+@Service
+public class OrderService implements OrderController {
+    private final OrderRepository orderRepository;
```

Comment:
You can use @Autowired to inject the bean. No need to create a constructor.

Reply:
Constructor injections is preferred over field injection.

https://stackoverflow.com/questions/40620000/spring-autowire-on-properties-vs-constructor